common-travel.lic

```diff
@@ -118,13 +115,25 @@ module DRCT
 
     timer = Time.now
     prev_room = XMLData.room_description + XMLData.room_title
+
+    # Moved flag declaration from the start of the method to here
+    # so that they only exist when needed and because the above code
+    # has lots of 'return' statements that'd have to be refactored to
+    # properly delete the flags at each method exit point.
+    # I didn't want to tackle that endeavor in this change.
+    Flags.add('travel-closed-shop', 'The door is locked up tightly for the night', 'You smash your nose', '^A servant (blocks|stops)')
+    Flags.add('travel-engaged', 'You are engaged')
+
     while Script.running.include?(script_handle)
       # Shop closed, stop script and open door then restart
       if Flags['travel-closed-shop']
         Flags.reset('travel-closed-shop')
         kill_script(script_handle)
         if /You open/ !~ DRC.bput('open door', 'It is locked', 'You .+', 'What were')
-          return false
+          # You cannot get to where you want to go,
+          # and no amount of retries will get you through the locked door.
+          restart_on_fail = false
+          break
```

Comment:
To still hit the code at the end of this method that deletes the flags, I changed the `return` to just don't restart on fail and then break out of the loop. Line 171 will now _not_ execute (retry movement) and the method will still return false because you're not at your destination.